controllers/sonataflow_controller.go

```diff
@@ -95,7 +101,7 @@ func (r *SonataFlowReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return reconcile.Result{}, nil
 	}
 
-	return profiles.NewReconciler(r.Client, workflow).Reconcile(ctx, workflow)
+	return factory.NewReconciler(r.Client, buildKnDiscoveryClient(r.Config), workflow).Reconcile(ctx, workflow)
```

Comment:
Pass only `r.Config` here so that in the future we can create other specific clients if needed such as `OpenShift` or `SonataFlow`. 

Also, we encapsulate the creation of KnativeClients into modules where we need them. wdyt?